core/src/main/scala/org/http4s/EntityEncoder.scala

```diff
@@ -147,11 +147,23 @@ trait EntityEncoderInstances extends EntityEncoderInstances0 {
   implicit val byteArrayEncoder: EntityEncoder[Array[Byte]] =
     chunkEncoder.contramap(Chunk.bytes)
 
-  // TODO fs2 port this is where we miss ByteVector
-  /*
   implicit val byteBufferEncoder: EntityEncoder[ByteBuffer] =
-    chunkEncoder.contramap(ByteVector.view)
-   */
+    chunkEncoder.contramap { bb =>
+      val duplicate = bb.duplicate()
+      // This is triggered if puts have been done, then we want to flip to read
+      if (duplicate.remaining != duplicate.capacity()) {
+        duplicate.flip()
+      }
+      if (duplicate.hasArray) {
+        Chunk.bytes(duplicate.array, 0, duplicate.remaining)
+      } else {
+        // This will create a new copy of the array, maybe expensive
+        // Should we break this if the ByteBuffer is very large?
```

Comment:
On the topic of files and `ByteBuffer`s, [`MappedByteBuffer`](https://docs.oracle.com/javase/7/docs/api/java/nio/MappedByteBuffer.html) is both interesting and offers some challenges since it will do blocking operations in our non-blocking framework. I don't have any great suggestions, just wanted to make sure everyone was aware of it.